compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -74,57 +74,75 @@ object DesugarEnums {
     else if (isEnumCase(cdef)) cdef.withMods(cdef.mods.withFlags(cdef.mods.flags | Final))
     else cdef
 
+  private def valuesDotTerm(name: TermName)(implicit src: SourceFile) =
+    Select(Ident(nme.DOLLAR_VALUES), name)
+
   private def valuesDot(name: String)(implicit src: SourceFile) =
-    Select(Ident(nme.DOLLAR_VALUES), name.toTermName)
+    valuesDotTerm(name.toTermName)
+
   private def registerCall(implicit ctx: Context): List[Tree] =
     if (enumClass.typeParams.nonEmpty) Nil
     else Apply(valuesDot("register"), This(EmptyTypeIdent) :: Nil) :: Nil
 
   /**  The following lists of definitions for an enum type E:
    *
    *   private val $values = new EnumValues[E]
```

Comment:
We can leave it like this for now, but I'd like to get rid of `EnumValues` altogether. The idea would be that code
for enums can be generated without reference to collection classes. Let's create an issue so that we can come back to this.